elfo/tests/update_config.rs

```diff
@@ -0,0 +1,83 @@
+#![cfg(feature = "test-util")]
+
+use elfo::{config::AnyConfig, messages::ConfigUpdated, prelude::*};
+use elfo_core::messages::{ConfigRejected, UpdateConfig};
+use serde::Deserialize;
+use serde_value::Value;
+use toml::toml;
+
+#[tokio::test]
+async fn singleton_actor_update_config() {
+    #[message]
+    struct StartSingleton;
+
+    #[message(ret = usize)]
+    struct GetLimit;
+
+    #[derive(Debug, Clone, Deserialize)]
+    struct Config {
+        limit: usize,
+    }
+
+    let blueprint = ActorGroup::new()
+        .config::<Config>()
+        .exec(move |mut ctx| async move {
+            while let Some(envelope) = ctx.recv().await {
+                msg!(match envelope {
+                    StartSingleton | ConfigUpdated => continue,
+                    (GetLimit, token) => {
+                        ctx.respond(token, ctx.config().limit);
+                    }
+                    _ => unreachable!(),
+                });
+            }
+        });
+
+    let config = toml! {
+        limit = 128
+    };
+    let proxy = elfo::test::proxy(blueprint, config).await;
+
+    proxy.send(StartSingleton).await;
+    assert_eq!(proxy.request(GetLimit).await, 128);
+
+    // Update with valid config through message.
+    let config = AnyConfig::new(
```

Comment:
I believe it should be `AnyConfig::deserialize(toml! {..})`.

Moreover, It would be nice to rename `fn new` to `from_value` and mark it as `#[stability::unstable]` because it's an implementation detail